### PR TITLE
Added more details to define Graphql schema for better understanding

### LINKED ISSUE #7580: Add details to Define the GraphQL schema for a module
# General issue

The lack of examples in some sections make this page uninformative.

Path to file: /guides/v2.4/graphql/develop/create-graphqls-file.html

## Description:

There are no examples under "Declare input attributes", "Specify output attributes", "Define the output interface", "Define enumerations"
It makes these sections nearly useless to people not already familiar with Magento's integration of GraphQl - it just says what is possible to do but doesn't show how to do it. 

## Possible solutions:

Link to a schema file in some module which has a good example of an implementation of whatever the section is about.
Add an example in the page itself, such as in the "Define queries" section

### PR DESCRIPTION
**Pull request by sathiyaaa**
## Purpose of this pull request

This pull request (PR) fixes #7580 

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/graphql/develop/create-graphqls-file.html
-  https://devdocs.magento.com/guides/v2.3/graphql/develop/create-graphqls-file.html

whatsnew
Added detailed examples to [Define the GraphQL schema for a module](https://devdocs.magento.com/guides/v2.4/graphql/develop/create-graphqls-file.html).

### CODE REVIEW

File: src/guides/v2.4/graphql/develop/create-graphqls-file.md [hunk 1]
```diff
@@ -50,18 +50,144 @@ If all your module's attributes are extension attributes for existing modules, t
...
+    }
+    page_info {
+      page_size
+      current_page
+    }
+  }
+}
+```
+
+The search returns 45 items, but only the first two items are returned on the current page and all the information regarding the page is returned.
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/develop/create-graphqls-file.md:167]: Same question as earlier

File: src/guides/v2.4/graphql/develop/create-graphqls-file.md [hunk 2]
```diff
@@ -50,18 +50,144 @@ If all your module's attributes are extension attributes for existing modules, t
...
 
 Output types that represent entities that can be manipulated (created, updated, or removed) and/or can be cached on the client MUST have `id` field. The type of the field SHOULD be `ID`.
 
+The following example shows the `products` query. The `page_info` attribute contains the `SearchResultPageInfo` data type which is defined in the `schema.graphqls` file under `ModuleGraphQl`. In other use cases, you would be required to create your own output type in `<magento_root>/app/code/<vendor_name>/<module_name>/etc/schema.graphqls` file.
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/develop/create-graphqls-file.md:138]:
```suggestion
The following example shows the `products` query. The `page_info` attribute contains the `SearchResultPageInfo` data type which is defined in the `schema.graphqls` file under `ModuleGraphQl`. In other use cases, you would be required to create your own output type in the `<magento_root>/app/code/<vendor_name>/<module_name>/etc/schema.graphqls` file.
```

File: src/guides/v2.4/graphql/develop/create-graphqls-file.md [hunk 3]
```diff
@@ -50,18 +50,144 @@ If all your module's attributes are extension attributes for existing modules, t
...
 You must know the data type of each attribute, whether it is scalar or an object, and whether it can be part of an array. In addition, each attribute within an object must be defined in the same manner.
 
 In a `schema.graphqls` file, the output `Interface` defines top-level attributes. Each object returned is defined in a `type` definition.
 
+The following example shows the `products` query. The query returns a `Products` object containing the attributes of the specified data types.
+
+Attribute | Data type | Description
+--- | --- | ---
+`aggregations` | [[Aggregation]]({{ page.baseurl }}/graphql/queries/products.html#Aggregation) | Layered navigation aggregations
+`filters` | LayerFilter | Deprecated. Use `aggregations` instead
```
Comment (BarnyShergold) [inline src/guides/v2.4/graphql/develop/create-graphqls-file.md:126]:
```suggestion
`filters` | LayerFilter | **Deprecated.** Use `aggregations` instead
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice piece of work!

File: src/guides/v2.3/graphql/develop/create-graphqls-file.md [hunk 4]
```diff
@@ -50,18 +50,146 @@ If all your module's attributes are extension attributes for existing modules, t
...
+    }
+    page_info {
+      page_size
+      current_page
+    }
+  }
+}
+```
+
+The search returns 45 items, but only the first two items are returned on the current page and all the information regarding the page is returned.
```
[the remark this answers is not in the document]
Comment (sathiyaaa) [reply src/guides/v2.3/graphql/develop/create-graphqls-file.md:169]: No, `total_count` is basically the total number of products that match the particular search criteria. If there is only one product which matches the search criteria then `page_size` would be 1 and the `total_count` would also be 1

[the remark this answers is not in the document]
Comment (sathiyaaa) [reply src/guides/v2.3/graphql/develop/create-graphqls-file.md:169]: rsz_page_size

**Review by BarnyShergold — APPROVED** [review]
Excellent!

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Pull request merged**